flink-python/pyflink/fn_execution/state_impl.py

```diff
@@ -810,6 +847,15 @@ def get_reducing_state(self, name, coder, reduce_function):
         self._all_states[name] = reducing_state
         return reducing_state
 
+    def get_aggregating_state(self, name, coder, agg_function):
```

Comment:
The most logic of `get_aggregating_state` , `get_value_state`, `get_reducing_state` and `get_list_state` are the same, maybe we can extract a method to reduce code duplication.